yarn-project/aztec/src/cli/cmds/start_faucet.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Faucet,
+  FaucetSchema,
+  createFaucetHttpServer,
+  faucetConfigMapping,
+  getFaucetConfigFromEnv,
+} from '@aztec/aztec-faucet';
+import { NamespacedApiHandlers } from '@aztec/foundation/json-rpc/server';
+import { LogFn } from '@aztec/foundation/log';
+
+import { extractNamespacedOptions, extractRelevantOptions } from '../util.js';
+
+export async function startFaucet(
+  options: any,
+  signalHandlers: (() => Promise<void>)[],
+  services: NamespacedApiHandlers,
+  log: LogFn,
+) {
+  const faucetOptions = extractNamespacedOptions(options, 'faucet');
+  const config = {
+    ...getFaucetConfigFromEnv(),
+    ...extractRelevantOptions(options, faucetConfigMapping, 'faucet'),
+  };
+
+  const faucet = await Faucet.create(config);
+  if (faucetOptions.apiServer) {
+    const httpServer = createFaucetHttpServer(faucet);
+    httpServer.listen(faucetOptions.apiServerPort);
+    signalHandlers.push(() => new Promise(res => httpServer.close(() => res())));
+    log(`Faucet now running on port: ${faucetOptions.apiServerPort}`);
+  }
```

Comment:
Would be nice to merge these two servers in one so that there's only one port to manage
#10581